parents/integrationtest/pom.xml

```diff
@@ -140,33 +140,56 @@
-                        <version>${test.elasticsearch.connection.version}</version>
-                        <timeout>90</timeout><!-- Make it work on our slow CI -->
-                        <!-- auto-create-index is necessary for non-OSS Elasticsearch -->
-                        <autoCreateIndex>true</autoCreateIndex>
-                        <environmentVariables>
-                            <JAVA_HOME>${test.elasticsearch.run.java_home}</JAVA_HOME>
-                        </environmentVariables>
+                        <images>
+                            <image>
+                                <name>docker.elastic.co/elasticsearch/elasticsearch:${test.elasticsearch.connection.version}</name>
+                                <alias>elasticsearch</alias>
+                                <run>
+                                    <env>
+                                        <discovery.type>single-node</discovery.type>
+                                        <!-- fix Docker images for older versions -->
+                                        <xpack.security.enabled>false</xpack.security.enabled>
+                                    </env>
+                                    <ports>
+                                        <port>9200:9200</port>
+                                        <port>9300:9300</port>
+                                    </ports>
+                                    <log>
+                                        <prefix>Elasticsearch: </prefix>
+                                        <date>default</date>
+                                        <color>cyan</color>
+                                    </log>
+                                    <wait>
+                                        <http>
+                                            <url>http://localhost:9200</url>
+                                            <method>GET</method>
+                                            <status>200</status>
+                                        </http>
+                                        <time>20000</time>
+                                    </wait>
+                                </run>
+                            </image>
+                        </images>
+                        <!--Stops all images currently running, not just those we just started.
+                          Useful to stop processes still running from a previously failed integration test run -->
+                        <allContainers>true</allContainers>
                     </configuration>
-                    <!-- Different executions from 2.x: the "start" goal has been renamed in version 5  -->
                     <executions>
                         <execution>
-                            <id>start-elasticsearch</id>
-                            <phase>pre-integration-test</phase>
+                            <id>docker-start</id>
+                            <phase>process-test-classes</phase>
```

Comment:
```suggestion
                            <phase>pre-integration-test</phase>
```

Reply:
Right, pre integration tests seems to be the right phase to trigger the docker start.
Anyway I'm asking why in Quarkus they use `process-test-classes`...
Probably there is some reason...

Reply:
Maybe they have some non-IT tests that need a container...

Reply:
Oh, don't get me started about this... They are basically hacking surefire to run integration tests...

Reply:
I imagined that :)